src/Symfony/Component/HttpKernel/EventListener/ExceptionListener.php

```diff
@@ -50,6 +57,17 @@ public function logKernelException(GetResponseForExceptionEvent $event)
 
     public function onKernelException(GetResponseForExceptionEvent $event)
     {
+        if (null === $this->controller) {
+            if (!$event->isMasterRequest()) {
+                return;
+            }
+            if (!$this->isTerminating) {
```

Comment:
for the record, the state managed by this property ensures that we don't break apps that handle exceptions outside of the stack:
```php
try {
    $kernel->handle($request);
} catch (SomeException $e) {
    // ...
}
```
as required in some framework migration strategies.
A hinted in the PR description, this allows running the fallback exception logic only when no catch block stopped the bubbling, in `terminateWithException`.